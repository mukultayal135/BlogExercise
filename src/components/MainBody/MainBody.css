main {
  width: 100%;
  display: flex;
  flex-grow: 1;
}
.cards-container {
  padding-bottom: 20px;
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  gap: 30px 50px;
}
@media (max-width: 910px) {
  .cards-container {
    justify-content: center;
  }
}
