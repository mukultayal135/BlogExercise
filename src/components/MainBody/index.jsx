import React, { useEffect, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import Card from '../Card';
import './MainBody.css';
import { GET_BLOG_DATA } from '../../constants/apiEndPoints';
import makeRequest from '../../utils/makeRequest';

const MainBody = () => {
  const [blogData, setBlogData] = useState();
  const navigate = useNavigate();

  useEffect(() => {
    makeRequest(GET_BLOG_DATA, navigate).then((response) => {
      setBlogData(response);
    });
  }, []);

  return (
    <main>
      {blogData ? (
        <div className="cards-container page-padding">
          {blogData.map((eachBlog) => (
            <Card key={eachBlog.id} blogData={eachBlog} />
          ))}
        </div>
      ) : (
        <div className="blogDataLoader">
          <p>Loading....</p>
        </div>
      )}
    </main>
  );
};

export default MainBody;
