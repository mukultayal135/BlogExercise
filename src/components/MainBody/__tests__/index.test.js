import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import MainBody from '..';
import makeRequest from '../../../utils/makeRequest';
import { mockBlogData } from '../../../mocks/blogData';

jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn(),
}));

jest.mock('../../../utils/makeRequest/');

describe('MainBody', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should show loading text when data is still loading', async () => {
    makeRequest.mockResolvedValue(mockBlogData);
    expect(makeRequest).not.toHaveBeenCalled();
    const { asFragment } = render(<MainBody />);
    expect(screen.getByText('Loading....')).toBeTruthy();
    await waitFor(() => {
      expect(screen.getByText('mock title 1')).toBeTruthy();
    });
    expect(asFragment()).toMatchSnapshot();
    expect(makeRequest).toHaveBeenCalledTimes(1);
  });
  it('should render all the cards when data is loaded', async () => {
    makeRequest.mockResolvedValue(mockBlogData);
    render(<MainBody />);

    await waitFor(() => {
      expect(screen.getAllByTestId('card').length).toEqual(2);
    });
  });
});
