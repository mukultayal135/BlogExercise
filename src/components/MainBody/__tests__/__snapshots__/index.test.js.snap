// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainBody should show loading text when data is still loading 1`] = `
<DocumentFragment>
  <main>
    <div
      class="cards-container page-padding"
    >
      <div
        class="card"
        data-testid="card"
      >
        <div>
          <img
            alt=""
            class="card-image"
            src="abstract.png"
          />
        </div>
        <div
          class="card-content card-padding"
        >
          <div
            class="date-time"
          >
            <div
              class="date"
            >
              29th
      June, 
      2021
            </div>
            <div
              class="read-time"
            >
              2 mins
            </div>
          </div>
          <div
            class="card-heading"
          >
            mock title 1
          </div>
          <div
            class="card-description"
          >
            mock description 1
          </div>
        </div>
        <hr />
        <div
          class="reactions"
        >
          <div
            class="clap"
          >
            <img
              alt="clapImage"
              src="clapping.svg"
            />
            <div
              class="count"
              data-testid="count-clap"
            >
              10
            </div>
          </div>
          <div
            class="heart"
          >
            <img
              alt="heartImg"
              src="heart-black.svg"
            />
          </div>
        </div>
      </div>
      <div
        class="card"
        data-testid="card"
      >
        <div>
          <img
            alt=""
            class="card-image"
            src="abstract.png"
          />
        </div>
        <div
          class="card-content card-padding"
        >
          <div
            class="date-time"
          >
            <div
              class="date"
            >
              29th
      June, 
      2022
            </div>
            <div
              class="read-time"
            >
              4 mins
            </div>
          </div>
          <div
            class="card-heading"
          >
            mock title 2
          </div>
          <div
            class="card-description"
          >
            mock description 2
          </div>
        </div>
        <hr />
        <div
          class="reactions"
        >
          <div
            class="clap"
          >
            <img
              alt="clapImage"
              src="clapping.svg"
            />
            <div
              class="count"
              data-testid="count-clap"
            >
              12
            </div>
          </div>
          <div
            class="heart"
          >
            <img
              alt="heartImg"
              src="heart-red.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
